network/event/peer_event.go

```diff
@@ -0,0 +1,39 @@
+package event
+
+import "github.com/libp2p/go-libp2p-core/peer"
+
+type PeerEventType uint
+
+const (
+	PeerConnected        PeerEventType = iota // Emitted when a peer connected
+	PeerFailedToConnect                       // Emitted when a peer failed to connect
+	PeerDisconnected                          // Emitted when a peer disconnected from node
+	PeerDialCompleted                         // Emitted when a peer completed dial
+	PeerAddedToDialQueue                      // Emitted when a peer is added to dial queue
+)
+
+var peerEventToName = map[PeerEventType]string{
```

Comment:
I think we can get rid of this map, if we implement the `String()` method to switch on the underlying value.

Reply:
I don't think that's wise and it's not scalable. @Kourin1996 Added this map as an easy fix if we need to add additional network events - that way we just update the map, and not add a new switch case. It also makes the code much cleaner

Reply:
Yes, but this map is used solely for the purpose of fetching a string format (only used in `String()` method). In order to scale with this approach, you would just need to implement another `case` branch. If this map had any other use cases, then we could discuss keeping it in memory (for the entire duration of the program).

Reply:
@Kourin1996 what do you think?